File: backend/app/models/student.py
from datetime import date, datetime

from sqlalchemy import Date, DateTime, Enum, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.enums import StudentStatus
from app.database.base import Base


class Student(Base):
    __tablename__ = "students"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    cpf: Mapped[str] = mapped_column(String(14), unique=True, index=True, nullable=False)
    birth_date: Mapped[date] = mapped_column(Date, nullable=False)
    phone: Mapped[str] = mapped_column(String(32), nullable=False)
    email: Mapped[str] = mapped_column(String(320), unique=True, index=True, nullable=False)
    address: Mapped[str] = mapped_column(Text, nullable=False)
    status: Mapped[StudentStatus] = mapped_column(
        Enum(StudentStatus, native_enum=False, length=20),
        nullable=False,
        default=StudentStatus.ACTIVE,
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

File: frontend/src/types/student.ts
import type { PageQueryParams, Status } from "./common";

export type FinancialStatus =
  | "IN_GOOD_STANDING"
  | "DEFAULTER"
  | "NO_ACTIVE_ENROLLMENT"
  | "INACTIVE";

export interface Student {
  id: number;
  name: string;
  cpf: string;
  birth_date: string;
  phone: string;
  email: string;
  address: string;
  status: Status;
  financial_status: FinancialStatus;
  created_at: string;
  updated_at: string;
}

export interface StudentPayload {
  name: string;
  cpf: string;
  birth_date: string;
  phone: string;
  email: string;
  address: string;
  status: Status;
}

export interface StudentQueryParams extends PageQueryParams {
  search?: string;
  cpf?: string;
  email?: string;
  name?: string;
  status?: Status;
}
